deltachat-rpc-server/src/bin/deltachat-rpc-server/main.rs

```diff
@@ -0,0 +1,72 @@
+///! Delta Chat core RPC server.
+///!
+///! It speaks JSON Lines over stdio.
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Result;
+use deltachat_jsonrpc::api::events::event_to_json_rpc_notification;
+use deltachat_jsonrpc::api::{Accounts, CommandApi};
+use futures_lite::stream::StreamExt;
+use tokio::io::{self, AsyncBufReadExt, BufReader};
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+use yerpc::{RpcClient, RpcSession};
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<()> {
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
+
+    let path = std::env::var("DC_ACCOUNTS_PATH").unwrap_or_else(|_| "./accounts".to_string());
+    log::info!("Starting with accounts directory `{}`.", path);
+    let accounts = Arc::new(RwLock::new(
+        Accounts::new(PathBuf::from(&path)).await.unwrap(),
+    ));
+
+    log::info!("Creating JSON-RPC API.");
+    let state = CommandApi::from_arc(accounts.clone());
+
+    let (client, mut out_receiver) = RpcClient::new();
+    let session = RpcSession::new(client.clone(), state);
+
+    // Events task converts core events to JSON-RPC notifications.
+    let events_task: JoinHandle<Result<()>> = tokio::spawn(async move {
+        let events = accounts.read().await.get_event_emitter();
```

Comment:
you can avoid acquiring the read lock by extracting the event emitter before the lock is created.  but it's more a matter of style than anything else.  i do think it's clearer that way though, because the accounts would be moved to the CommandApi and the emitter would be moved to this task.  because the current way i had to scan around to figure out what else was happening to accounts.